lib/Controller/RssController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Nextcloud - News
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author    Paul Tirk <paultirk@paultirk.com>
+ * @copyright 2020 Paul Tirk
+ */
+
+namespace OCA\News\Controller;
+
+use FeedIo\Feed;
+use FeedIo\FeedIo;
+use OCA\News\Service\ItemServiceV2;
+use OCA\News\Utility\XMLResponse;
+use OCP\AppFramework\Http;
+use \OCP\IRequest;
+use OCP\IUserManager;
+use \OCP\IUserSession;
+use Psr\Log\LoggerInterface;
+
+/**
+ * This controller is in charge of exposing RSS feeds from
+ * gathered feed items.
+ */
+class RssController extends Controller
+{
+    use JSONHttpErrorTrait;
+
+    /**
+     * @var ItemServiceV2
+     */
+    private $itemService;
+
+    /**
+     * @var IUserManager
+     */
+    private $userManager;
+
+    /**
+     * @var FeedIo
+     */
+    private $feedIo;
+
+    public function __construct(
+        IRequest      $request,
+        IUserSession  $userSession,
+        IUserManager  $userManager,
+        ItemServiceV2 $itemService,
+        FeedIo        $feedIo
+    ) {
+        parent::__construct($request, $userSession);
+
+        $this->userManager = $userManager;
+        $this->itemService = $itemService;
+        $this->feedIo = $feedIo;
+    }
+
+    /**
+     * @NoCSRFRequired
+     * @PublicPage
+     *
+     * @param string $name
+     * @return array|mixed|\OCP\AppFramework\Http\JSONResponse
+     */
+    public function starred($userId)
```

Comment:
Yes, I was wondering if a token is needed but decided to go with a true, public feed. Do you feel it can bring too many security risks or confusion?

Reply:
Personally, I wouldn't want everyone on the Internet to be able to access my starred items.
For example, you may have subscribed to a medical feed and starred the articles matching your condition to save them for later reading.

Even without guessing the user id (which isn't that hard for many users who use the same everywhere possible), there are NC servers were the usernames are visible are visible to other users or can be found via the Profiles feature.

Reply:
Another issue for many feeds if not all is that at least in Germany but probably also in a lot of other States you are not allowed to re-publish the content openly. You would need to ask for permission for each feed/item.

Just because the content is public it doesn't mean that you have the right to publish it.

If you do it publicly it might get detected by content scanners.

Reply:
I agree, maybe both should be stated in the user interface:
- the feature will publicly repost contents, there is no authentication layer
- user should have the right to repost articles

Maybe the endpoint should also be denied in robots.txt.

Do you want me to check what's displayed in TinyTinyRSS today or adding those mitigations would be enough?

Reply:
Neither of those are mitigations though since there are a lot of custom UIs for the news app. I would say that it's only acceptable with authentication.

Reply:
Thanks for your answer. Would generating a secret key for the feed be acceptable instead of having full user authentication? It would allow sharing the feed publicly without having to share your own credentials.

Reply:
If you do full user authentication you can generate an app password which is mostly the same except would not require the news team to maintain it.

Reply:
But if the app password is leaked, it gives full access to the user account right?
So if I share the link to someone, I'm giving full access to my account.

Reply:
Yes, but the way you're suggesting the news team needs to maintain it's own authentication mechanism. Which I at the very least refuse to do.

Reply:
Thanks, I will then close this draft pull request. My understanding from this discussion is that there is no way to implement a feature that would answer the following need: ability to easily share a feed of stared items, with or without a password, to a tier person that does not necessarily have a NextCloud account.

It is super sad because this feature (having a public RSS feed for favorite items) exist in at least the two biggest alternatives of NextCloud News. Namely: Tiny Tiny RSS and FreshRSS (you can test the feature on their demo servers).